homeassistant/components/aidot/light.py

```diff
@@ -0,0 +1,225 @@
+        self._attr_name = device["name"]
+        self.pingtask = None
+        self.recvtask = None
+
+        modelId = device["modelId"]
+        manufacturer = modelId.split(".")[0]
+        model = modelId[len(manufacturer) + 1 :]
+        mac = format_mac(device["mac"]) if device["mac"] is not None else ""
+        identifiers: set[tuple[str, str]] = (
+            set({(DOMAIN, self._attr_unique_id)}) if self._attr_unique_id else set()
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers=identifiers,
+            connections={(CONNECTION_NETWORK_MAC, mac)},
+            manufacturer=manufacturer,
+            model=model,
+            name=device["name"],
+            hw_version=device["hardwareVersion"],
+        )
+        self._cct_min = 0
+        self._cct_max = 0
+
+        supported_color_modes = set()
+        if "product" in device and "serviceModules" in device["product"]:
+            for service in device["product"]["serviceModules"]:
+                if service["identity"] == "control.light.rgbw":
+                    supported_color_modes.add(ColorMode.RGBW)
+                elif service["identity"] == "control.light.cct":
+                    self._cct_min = int(service["properties"][0]["minValue"])
+                    self._cct_max = int(service["properties"][0]["maxValue"])
+                    supported_color_modes.add(ColorMode.COLOR_TEMP)
+                # elif "control.light.effect.mode" == service["identity"]:
+                # self._attr_supported_features = LightEntityFeature.EFFECT | LightEntityFeature.FLASH
+                # allowedValues = service["properties"][0]["allowedValues"]
+                # print(f"allowedValues: {allowedValues}")
+                # self._attr_effect_list = [item["name"] for item in allowedValues]
+
+        if ColorMode.RGBW in supported_color_modes:
+            self._attr_color_mode = ColorMode.RGBW
+            self._attr_supported_color_modes = {ColorMode.RGBW, ColorMode.COLOR_TEMP}
+        elif ColorMode.COLOR_TEMP in supported_color_modes:
+            self._attr_color_mode = ColorMode.COLOR_TEMP
+            self._attr_supported_color_modes = {ColorMode.COLOR_TEMP}
+        else:
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+        self.lanCtrl = Lan(device, user_info)
+        self.lanCtrl.setUpdateDeviceCb(self.updateState)
+
+        async def handle_event(event):
+            if not self.lanCtrl.connecting and not self.lanCtrl.connectAndLogin:
+                await self.lanCtrl.connect(event.data["ipAddress"])
+                self.pingtask = hass.loop.create_task(self.lanCtrl.ping_task())
+                self.recvtask = hass.loop.create_task(self.lanCtrl.recvData())
+
+        hass.bus.async_listen(device["id"], handle_event)
+
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.release)
+
+    async def release(self, event: Event):
```

Comment:
Cleanup should not happen when the `EVENT_HOMEASSISTANT_STOP` is fired, it should happen when the entity is removed.
To make that happen:
```
self.async_on_remove(self.release)
```